File: Data Structures/Linked List/Singly Linked List/SinglyLinkedList.cpp
#include <iostream>

class Node
{
private:
    int data;
    Node *next;

public:
    Node()
    {
        data = 0;
        this->next = nullptr;
    }

    Node(int data)
    {
        this->data = data;
        this->next = nullptr;
    }

    int getData() const
    {
        return data;
    }

    void setData(int data)
    {
        this->data = data;
    }

    void insertEnd(Node *&head, int data)
    {
        Node *newNode = new Node(data);

        if (head == nullptr)
        {
            head = newNode;
        }

        Node *current = head;
        while (current->next != nullptr)
        {
            current = current->next;
        }
        current->next = newNode;
    }

    void display(Node *head)
    {
        Node *current = head;
        while (current != nullptr)
        {
            std::cout << current->data << " ";
            current = current->next;
        }
    }
};

int main()
{
    Node *head = nullptr;
    Node node;
    node.insertEnd(head, 1);
    node.insertEnd(head, 2);
    node.insertEnd(head, 3);
    node.insertEnd(head, 4);
    node.insertEnd(head, 5);

    node.display(head);
}

File: Data Structures/Linked List/Singly Linked List/SinglyLinkedList.java
public class SinglyLinkedList {
    private ListNode head;

    private static class ListNode {
        private int data;
        private ListNode next;

        public ListNode(int data) {
            this.data = data;
            this.next = null;
        }
    }

    public void TraversalLinkedList(ListNode head) {
        ListNode current = head;

        while (current != null) {
            System.out.println(current.data + " ");
            current = current.next;
        }
    }

    public boolean SearchingLinkedList(ListNode head, int target) {
        ListNode current = head;

        while (current != null) {
            if (current.data == target) {
                return true;
            }
            current = current.next;
        }

        return false;
    }

    public int FindingLength(ListNode head) {
        if (head == null) {
            return 0;
        }

        ListNode current = head;
        int count = 0;
        while (current != null) {
            count++;
            current = current.next;
        }

        return count;
    }

    public void InsertAtBeginning(int data) {
        ListNode newNode = new ListNode(data);
        newNode.next = head;
        head = newNode;

    }

    public static void main(String[] args) {
        SinglyLinkedList sll = new SinglyLinkedList();
        sll.head = new ListNode(10);
        ListNode second = new ListNode(1);
        ListNode third = new ListNode(15);
        ListNode fourth = new ListNode(4);

        sll.head.next = second;
        second.next = third;
        third.next = fourth;

        sll.InsertAtBeginning(2);
        sll.TraversalLinkedList(sll.head);

    }
}
